site/gatsby-site/src/components/cite/TsneVisualization.js

```diff
@@ -588,3 +565,13 @@ var Visualization = styled.div`
     z-index: 2;
   }
 `;
+
+import React, { useState, useEffect } from 'react';
```

Comment:
It's unorthodox, but I think it's the correct place for them. When you open a file, which do you think is more useful as the first thing you see: the signature of the default function, or `import React from 'react';` ? In low-level languages, you need to define everything before it's used for the benefit of the compiler, but in a high-level, dynamic language like JavaScript, you can order declarations for the benefit of the reader. 

I understand if this is too weird to be included, I just wanted to give it a try.

Reply:
I agree with the readability part, but to do this properly, we should do it on every file and update the linter settings, etc. 

Also, even if it works, technically, it is not valid javascript (on vanilla js it will fail)

I don't know if you use vscode, but I've found this as a consolation prize:

![image](https://user-images.githubusercontent.com/1172479/198149566-b0e43d2d-87a9-46b5-8917-acfe090aef55.png)

Reply:
> I agree with the readability part, but to do this properly, we should do it on every file and update the linter settings, etc.

Yeah, you're right. It's probably too much effort to be worth it. I've put them back on top.

> Also, even if it works, technically, it is not valid javascript (on vanilla js it will fail)

Which part isn't valid? When I load this in the browser it displays "Hello, World!" even with the import on the bottom:

```html
<!doctype html>
<title>Title</title>
<script type="module" src="helloWorld.js"></script>
<script type="module" src="main.js"></script>
```

```js
// helloWorld.js
export default function helloWorld() {
  document.body.innerHTML = 'Hello, World!';
}
```
```js
// main.js
helloWorld();
import helloWorld from "./helloWorld.js";
```

> I don't know if you use vscode, but I've found this as a consolation prize:

I don't, but thanks for the idea! I came up with a vimrc one-liner to replicate it:

```vim
autocmd Filetype javascript :let @/ = "import" | silent! :execute "normal vN zf"
```

Reply:
Oh, so it hoists imports too... feels so wrong 😆